spock-core/src/main/java/org/spockframework/runtime/extension/builtin/IncludeExcludeExtension.java

```diff
@@ -14,16 +14,20 @@
 
 package org.spockframework.runtime.extension.builtin;
 
-import org.spockframework.runtime.extension.AbstractGlobalExtension;
+import org.spockframework.runtime.extension.*;
 import org.spockframework.runtime.model.*;
 import spock.config.*;
 
 import java.lang.annotation.Annotation;
 import java.util.List;
 
 @SuppressWarnings("UnusedDeclaration")
-public class IncludeExcludeExtension extends AbstractGlobalExtension {
-  private RunnerConfiguration config;
+public class IncludeExcludeExtension implements IGlobalExtension {
+  private final RunnerConfiguration config;
+
+  public IncludeExcludeExtension(RunnerConfiguration config) {
```

Comment:
Is there any specific reason you use constructor injection in all built-in extensions instead of keeping field injection?
Imho field injection is cleaner, more concise and more readable if the object is needed outside the constructor.
I personally only use constructor injection for thing that I actually need inside the constructor.

Reply:
re CI: You don't need to make the field accessible anymore, can declare it as final, and it is obvious what is a dependency. 

As for why do it for every built-in extension: consistency.

Reply:
> As for why do it for every built-in extension: consistency.
Of course, the question was not "why not only for some", but "why at all"

But yeah, without using explicit `@Inject` annotation it indeed might make it clearer it gets injected.